File: user-rest-service-1.0.2/generated-code/jaxrs/src/main/java/io/swagger/api/UserApi.java
package io.swagger.api;

import io.swagger.model.*;

import com.wordnik.swagger.annotations.ApiParam;

import com.sun.jersey.multipart.FormDataParam;

import io.swagger.model.User;


import java.util.List;
import io.swagger.api.NotFoundException;

import java.io.InputStream;

import com.sun.jersey.core.header.FormDataContentDisposition;
import com.sun.jersey.multipart.FormDataParam;

import javax.ws.rs.core.Response;
import javax.ws.rs.*;

@Path("/user")
@com.wordnik.swagger.annotations.Api(value = "/user", description = "the user API")

public class UserApi {
  
  @GET
  
  @Consumes({ "application/json",  })
  @Produces({ "*/*" })
  @com.wordnik.swagger.annotations.ApiOperation(value = "read User by ID", notes = "pass ID to read user", response = User.class)
  @com.wordnik.swagger.annotations.ApiResponses(value = { 
    @com.wordnik.swagger.annotations.ApiResponse(code = 404, message = "User not found"),
    
    @com.wordnik.swagger.annotations.ApiResponse(code = 200, message = "success") })

  public Response getUser(@ApiParam(value = "id",required=true) @QueryParam("id") Long id)
      throws NotFoundException {
      // do some magic!
      return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "magic!")).build();
  }

  
  @POST
  
  @Consumes({ "application/json",  })
  @Produces({ "*/*" })
  @com.wordnik.swagger.annotations.ApiOperation(value = "create or update a user name by id", notes = "saveUser", response = User.class)
  @com.wordnik.swagger.annotations.ApiResponses(value = { 
    @com.wordnik.swagger.annotations.ApiResponse(code = 200, message = "success") })

  public Response saveUser(@ApiParam(value = "id",required=true) @QueryParam("id") Long id,
    @ApiParam(value = "name",required=true) @QueryParam("name") String name)
      throws NotFoundException {
      // do some magic!
      return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "magic!")).build();
  }

  
}

File: user-rest-service-1.0.2/generated-code/scalatra/src/main/scala/com/wordnik/client/api/UserApi.scala
package com.wordnik.client.api

import com.wordnik.client.model.User
import com.wordnik.client.model.Void


import java.io.File

import org.scalatra.{ TypedParamSupport, ScalatraServlet }
import org.scalatra.swagger._
import org.json4s._
import org.json4s.JsonDSL._
import org.scalatra.json.{ JValueResult, JacksonJsonSupport }
import org.scalatra.servlet.{FileUploadSupport, MultipartConfig, SizeConstraintExceededException}

import scala.collection.JavaConverters._

class UserApi (implicit val swagger: Swagger) extends ScalatraServlet 
    with FileUploadSupport
    with JacksonJsonSupport
    with SwaggerSupport {
  protected implicit val jsonFormats: Formats = DefaultFormats

  protected val applicationDescription: String = "UserApi"
  override protected val applicationName: Option[String] = Some("User")

  before() {
    contentType = formats("json")
    response.headers += ("Access-Control-Allow-Origin" -> "*")
  }


  

  val getUserOperation = (apiOperation[User]("getUser")
      summary "read User by ID"
      parameters(
        queryParam[Long]("id").description("")
        
        
        
        
        
        
        )
  )

  get("/user",operation(getUserOperation)) {
    
    
    
    

    
      
      val id = params.getAs[Long]("id")
      
    

    

    

    
    
    println("id: " + id)
  
  }


  

  val saveUserOperation = (apiOperation[User]("saveUser")
      summary "create or update a user name by id"
      parameters(
        queryParam[Long]("id").description("")
        
        
        
        
        
        ,
        queryParam[String]("name").description("")
        
        
        
        
        
        
        )
  )

  post("/user",operation(saveUserOperation)) {
    
    
    
    

    
      
      val id = params.getAs[Long]("id")
      
    

    

    

    
    
    println("id: " + id)
  
    
    
    

    
      
      val name = params.getAs[String]("name")
      
    

    

    

    
    
    println("name: " + name)
  
  }



}